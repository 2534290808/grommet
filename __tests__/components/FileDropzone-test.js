// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import FileDropzone from '../../src/js/components/FileDropzone';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('FileDropzone', () => {

  it('has correct default options', () => {
    const component = renderer.create(
      <FileDropzone />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a snapshot with label props', () => {
    const component = renderer.create(
      <FileDropzone label="Testing" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a snapshot with preview prop set to false', () => {
    const component = renderer.create(
      <FileDropzone preview={false} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a snapshot with multiple prop set to true', () => {
    const component = renderer.create(
      <FileDropzone multiple={true} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a snapshot with full screen drop target set to true', () => {
    const component = renderer.create(
      <FileDropzone fullDropTarget={true} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts a function as onDOMChange prop', () => {
    const component = renderer.create(
      <FileDropzone onDOMChange={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
