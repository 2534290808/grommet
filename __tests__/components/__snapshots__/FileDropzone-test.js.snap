exports[`FileDropzone accepts a function as onDOMChange prop 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Click or drop a file to upload
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={false}
    onChange={[Function]}
    type="file" />
</div>
`;

exports[`FileDropzone has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Click or drop a file to upload
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={false}
    onChange={[Function]}
    type="file" />
</div>
`;

exports[`FileDropzone should create a snapshot with full screen drop target set to true 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Click or drop a file to upload
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={false}
    onChange={[Function]}
    type="file" />
</div>
`;

exports[`FileDropzone should create a snapshot with label props 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Testing
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={false}
    onChange={[Function]}
    type="file" />
</div>
`;

exports[`FileDropzone should create a snapshot with multiple prop set to true 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Click or drop a file to upload
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={true}
    onChange={[Function]}
    type="file" />
</div>
`;

exports[`FileDropzone should create a snapshot with preview prop set to false 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-medium"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    aria-label="Box"
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--clickable grommetux-dropzone"
    id={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="group"
    style={Object {}}
    tabIndex={0}>
    <label
      className="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
      htmlFor={undefined}>
      Click or drop a file to upload
    </label>
  </div>
  <input
    className="grommetux-dropzone__input"
    multiple={false}
    onChange={[Function]}
    type="file" />
</div>
`;
